### PR TITLE
Add support for 'assert_not_equal'.

### LINKED ISSUE #31: assert_notequal missing?
I was a bit suprised that **assert_notequal** is missing.

### PR DESCRIPTION
**Pull request by jwheeler31**
This is essentially a copy/paste of `assert_equal`. However, unlike `assert_equal`, I think it's more correct to perform the equivalence test surrounding each operand in double quotes (see [assert_equal.bash#35](https://github.com/bats-core/bats-assert/blob/master/src/assert_equal.bash#L35) vs [assert_not_equal.bash#35](https://github.com/bats-core/bats-assert/blob/38bfbf61443c046acf91b678a5a8eb5a0ab13000/src/assert_not_equal.bash#L35)).

This would also close #31 .

### CODE REVIEW

File: test/assert_not_equal.bats [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  run assert_not_equal foo bar
+  assert_test_pass
+
+  run assert_not_equal "foo" "bar"
+  assert_test_pass
+
+  run assert_not_equal "foo" ""
+  assert_test_pass
+
+  run assert_not_equal "" "foo"
```
Comment (martin-schulze-vireso) [inline test/assert_not_equal.bats:15]: Missing assertion

File: README.md [hunk 2]
```diff
@@ -127,6 +128,28 @@ actual   : have
...
+  assert_not_equal 'foobar' 'foobar'
+}
+```
+
+On failure, the expected and actual values are displayed.
+
+```
+-- values should not be equal --
+unexpected : foobar
+actual     : foobar
```
Comment (martin-schulze-vireso) [inline README.md:146]: Do we really need the same value to be printed twice?

**Review by martin-schulze-vireso — COMMENTED** [review]
Minor nits

File: README.md (hunk 2 above)
Comment (jwheeler31) [reply README.md:146]: @martin-schulze-vireso, I think it's a matter of opinion. `assertpy` does reports this way:
```
def test_dummy_test() -> None:
>       assert_that(1 + 1).is_not_equal_to(2)
E       AssertionError: Expected <2> to be not equal to <2>, but was.
```
while `assertj` does it this way:
```
java.lang.AssertionError: 
Expecting actual:
  2
not to be equal to:
  2
```

While I'm not advocating that _bats-assert_ should be "just like everybody else", I do think that it should feel somewhat familiar, as well as follow the other patterns in the existing code base.

Comment (martin-schulze-vireso) [reply README.md:146]: I guess regardless of language the question is: does the comparison equal output string comparison or can the string values differ for something that is considered equal?

Comment (jwheeler31) [reply README.md:146]: > does the comparison equal output string comparison or can the string values differ for something that is considered equal?


If I'm understanding the question correctly, `assert_not_equal` only throws if the two values are exactly equal (as defined by `[[ "$1" == "$2" ]]`).

Comment (martin-schulze-vireso) [reply README.md:146]: Yes, I can't see how there will be a difference between the comparison and the string values. However, bash has always been good for surprises in weird corner cases. We can leave the code as is. I was just thinking about lengthy texts being printed twice.

Comment (jwheeler31) [reply README.md:146]: Sounds good.

Speaking of corner cases, I just discovered an infinite loop if you run the following:
```
run assert_equal "" ""
assert_test_fail
```

I have to dig some more before reporting it, but gotta love surprises.

File: test/assert_not_equal.bats (hunk 1 above)
Comment (jwheeler31) [reply test/assert_not_equal.bats:15]: Fixed.

**Review by martin-schulze-vireso — APPROVED** [review]

**Comment by jwheeler31** [comment]
@martin-schulze-vireso First open source PR! 🍻
